### DISCUSSION TITLE
Junos match multipatern does not work.

### DISCUSSION THREAD

**Discussion by eradek** [napalm-automation/napalm#1603] [Q&A]
I am trying to use device.cli to gather logs from Junos with multipatern matching however it returns no output. 

```
            logs = self.device.cli(['show interfaces ae2 extensive | match "Errors|flapped|Speed"'])
            print(logs)
```
Output:
router seems to be alive, gathering prechecks...!
{'show interfaces ae2 extensive | match "Errors|flapped|Speed"': ''}

Whereas running this directly on router:

router> show interfaces ae2 extensive | match "Errors|flapped|Speed"
  Link-level type: Ethernet, MTU: 1514, Speed: 700Gbps, BPDU Error: None, Ethernet-Switching Error: None,
  Last flapped   : 2022-03-30 10:18:59 UTC (1d 04:54 ago)
  Input errors:
    Errors: 0, Drops: 0, Framing errors: 0, Runts: 0, Giants: 0, Policed discards: 0, Resource errors: 0
  Output errors:
    Carrier transitions: 42, Errors: 0, Drops: 0, MTU errors: 0, Resource errors: 0

Does anyone has a clue?

**Comment by ktbyers** [comment]
I would remove all the pipe and match section and just try the straight command:

```
show interfaces ae2 extensive
```

Do you get a result back if you do that?

My first question is does Juniper's PyEZ interface/NETCONF allow you to send pipe characters and extra filters? Remember you are not interfacing to the CLI here, you are interfacing through the PyEZ library and NETCONF.

**Reply by eradek** [reply]
It doesn't look like pipe is the problem, its multiple arguments that cause it. If I run:

```
  logs = self.device.cli(['show interfaces ae2 extensive | match Errors'])
            print(logs)
```

it works fine.
